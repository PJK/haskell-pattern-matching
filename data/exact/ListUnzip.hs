module ListUnzip where

unzip :: [a] -> [(a, a)]
unzip [] = []
unzip (x:y:zz) = (x, y):unzip zz
-- List of odd length missing

