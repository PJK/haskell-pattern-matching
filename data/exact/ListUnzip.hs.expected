{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "unzip",
                {
                    "tag": "NonExhaustive",
                    "contents": [
                        [
                            [
                                {
                                    "tag": "InfixConstructorPattern",
                                    "contents": [
                                        {
                                            "tag": "WildcardPattern",
                                            "contents": []
                                        },
                                        ":",
                                        {
                                            "tag": "EmptyListPattern",
                                            "contents": []
                                        }
                                    ]
                                }
                            ],
                            ""
                        ]
                    ]
                }
            ]
        ],
        [
            [
                "unzip",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "EmptyListPattern",
                                "contents": []
                            }
                        ],
                        [
                            {
                                "tag": "EmptyListPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~b"
                                    },
                                    ":",
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~c"
                                    }
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "InfixConstructorPattern",
                                "contents": [
                                    {
                                        "tag": "WildcardPattern",
                                        "contents": []
                                    },
                                    ":",
                                    {
                                        "tag": "VariablePattern",
                                        "contents": "~c"
                                    }
                                ]
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
